packages/ember-data/lib/system/record-array-manager.js

```diff
@@ -101,6 +107,13 @@ export default Ember.Object.extend({
       filter = get(array, 'filterFunction');
       this.updateRecordArray(array, filter, typeClass, record);
     }, this);
+
+    if (this.liveRecordArrays.has(typeClass)) {
+      var liveRecordArray = this.liveRecordArrays.get(typeClass);
+      if (!liveRecordArray.isDestroyed && !liveRecordArray.isDestroying) {
```

Comment:
The check for the record array not being destroyed is needed as otherwise [this test](http://git.io/vk5zB) fails.

I am not satisfied about this check; how should this be tackled? Check if the `RecordArray` which is about to be destroyed is a `liveRecordArray` and if so, remove it from the map?